ghdlsynth build breaks

ghdlsynth uses the following to compile its interface module:
```
yosys-config --exec --cxx -c --cxxflags -o $@ $<
```

A recent change to yosys breaks the inclusion of rtlil.h... at line 380 it tries to include "constids.inc" which appears only in the kernel directory, and is not in the include path.  A quick hack to put the yosys/kernel directory in the path manually results in a working ghdl module.

I had a quick look at the changesets and it's not immediately obvious which change causes this, or if the build break is caused by an incomplete build or install issue, but it is in the last 2 days.  I have been tracking both ghdlsynth and yosys development, and this is a new bug.

Cheers, and thanks to all the Yosys developers.  In conjunction with the GHDL developer's work, very much progress has been made in just the last few weeks.

Not OP, but 2d0b4033 still doesn't work - the error doesn't occur during compilation of yosys, but when ghdl-yosys-plugin includes the yosys headers:

```
In file included from /usr/share/yosys/include/kernel/yosys.h:277,
                 from src/ghdl.cc:19:
/usr/share/yosys/include/kernel/rtlil.h:380:10: fatal error: constids.inc: No such file or directory
  380 | #include "constids.inc"
      |          ^~~~~~~~~~~~~~
```

I think the occurences in both [`rtlil.cc`](https://github.com/YosysHQ/yosys/blob/5d18fdadfe758fed14a93977d3e8e83096c26c5a/kernel/rtlil.cc#L45) and [`yosys.cc`](https://github.com/YosysHQ/yosys/blob/5d18fdadfe758fed14a93977d3e8e83096c26c5a/kernel/yosys.cc#L519) are fine, only [`rtlil.h`](https://github.com/YosysHQ/yosys/blob/5d18fdadfe758fed14a93977d3e8e83096c26c5a/kernel/rtlil.h#L380) needs to be changed.

--- a/kernel/rtlil.cc
+++ b/kernel/rtlil.cc
@@ -42,7 +42,7 @@ int RTLIL::IdString::last_created_idx_ptr_;
 #endif
 
 #define X(_id) IdString RTLIL::ID::_id;
-#include "constids.inc"
+#include "kernel/constids.inc"
 #undef X
 
 dict<std::string, std::string> RTLIL::constpad;

--- a/kernel/rtlil.h
+++ b/kernel/rtlil.h
@@ -377,7 +377,7 @@ namespace RTLIL
 
 	namespace ID {
 #define X(_id) extern IdString _id;
-#include "constids.inc"
+#include "kernel/constids.inc"
 #undef X
 	};
 

--- a/kernel/yosys.cc
+++ b/kernel/yosys.cc
@@ -516,7 +516,7 @@ void yosys_setup()
 	already_setup = true;
 
 #define X(_id) RTLIL::ID::_id = "\\" # _id;
-#include "constids.inc"
+#include "kernel/constids.inc"
 #undef X
 
 	#ifdef WITH_PYTHON

--- a/tests/various/.gitignore
+++ b/tests/various/.gitignore
@@ -3,3 +3,4 @@
 /write_gzip.v
 /write_gzip.v.gz
 /run-test.mk
+/plugin.so

--- a/tests/various/plugin.cc
+++ b/tests/various/plugin.cc
@@ -1,0 +1,15 @@
+#include "kernel/rtlil.h"
+
+YOSYS_NAMESPACE_BEGIN
+
+struct TestPass : public Pass {
+	TestPass() : Pass("test", "test") { }
+	void execute(std::vector<std::string> args, RTLIL::Design *design) YS_OVERRIDE
+	{
+		size_t argidx = 1;
+		extra_args(args, argidx, design);
+		log("Plugin test passed!\n");
+	}
+} TestPass;
+
+YOSYS_NAMESPACE_END

--- a/tests/various/plugin.sh
+++ b/tests/various/plugin.sh
@@ -1,0 +1,6 @@
+set -e
+rm -f plugin.so
+CXXFLAGS=$(../../yosys-config --cxxflags)
+CXXFLAGS=${CXXFLAGS// -I\/usr\/local\/share\/yosys\/include/ -I..\/..\/share\/include}
+../../yosys-config --exec --cxx ${CXXFLAGS} --ldflags -shared -o plugin.so plugin.cc
+../../yosys -m ./plugin.so -p "test" | grep -q "Plugin test passed!"

